diff --git a/os_dep/linux/os_intfs.c b/os_dep/linux/os_intfs.c
index a47d52a..f7cdcce 100644
--- a/os_dep/linux/os_intfs.c
+++ b/os_dep/linux/os_intfs.c
@@ -606,7 +606,13 @@ unsigned int rtw_classify8021d(struct sk_buff *skb)
 	return dscp >> 5;
 }
 
+#if (LINUX_VERSION_CODE<=KERNEL_VERSION(3,12,0))
 static u16 rtw_select_queue(struct net_device *dev, struct sk_buff *skb)
+#elif (LINUX_VERSION_CODE<=KERNEL_VERSION(3,13,0))
+static u16 rtw_select_queue(struct net_device *dev, struct sk_buff *skb, void *accel_priv)
+#else
+static u16 rtw_select_queue(struct net_device *dev, struct sk_buff *skb, void *accel_priv, select_queue_fallback_t fallback)
+#endif
 {
 	_adapter	*padapter = rtw_netdev_priv(dev);
 	struct mlme_priv *pmlmepriv = &padapter->mlmepriv;
