--- rtl8812au_linux-master/os_dep/linux/ioctl_cfg80211.c~	2017-07-31 23:54:36.000000000 +0200
+++ rtl8812au_linux-master/os_dep/linux/ioctl_cfg80211.c	2017-07-31 23:58:19.371143221 +0200
@@ -5991,7 +5991,11 @@
 #endif
 
 #if defined(CONFIG_PM) && (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 0, 0))
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 12, 0)
+	wiphy->max_sched_scan_reqs = 1;
+#else
 	wiphy->flags |= WIPHY_FLAG_SUPPORTS_SCHED_SCAN;
+#endif
 #ifdef CONFIG_PNO_SUPPORT
 	wiphy->max_sched_scan_ssids = MAX_PNO_LIST_COUNT;
 #endif
