--- rtl8812au_linux-master/os_dep/linux/rtw_android.c~	2016-06-19 17:56:30.000000000 +0200
+++ rtl8812au_linux-master/os_dep/linux/rtw_android.c	2016-06-19 17:57:23.699243076 +0200
@@ -574,7 +574,11 @@
 		goto exit;
 	}
 #ifdef CONFIG_COMPAT
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 6, 0))
 	if (is_compat_task()) {
+#else
+	if (in_compat_syscall()) {
+#endif
 		/* User space is 32-bit, use compat ioctl */
 		compat_android_wifi_priv_cmd compat_priv_cmd;
 
