--- rtl8812au_linux-master/os_dep/linux/ioctl_cfg80211.c.orig	2016-08-01 21:49:34.559616222 +0200
+++ rtl8812au_linux-master/os_dep/linux/ioctl_cfg80211.c	2016-08-01 21:50:12.690473128 +0200
@@ -25,6 +25,13 @@
 
 #include <rtw_wifi_regd.h>
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 7, 0))
+#define ieee80211_band nl80211_band
+#define IEEE80211_BAND_2GHZ NL80211_BAND_2GHZ
+#define IEEE80211_BAND_5GHZ NL80211_BAND_5GHZ
+#define IEEE80211_NUM_BANDS NUM_NL80211_BANDS
+#endif
+
 #define RTW_MAX_MGMT_TX_CNT (8)
 
 #define RTW_SCAN_IE_LEN_MAX      2304
--- rtl8812au_linux-master/os_dep/linux/wifi_regd.c.orig	2016-08-01 21:49:41.259767005 +0200
+++ rtl8812au_linux-master/os_dep/linux/wifi_regd.c	2016-08-01 21:50:02.516911448 +0200
@@ -10,6 +10,13 @@
 
 #include <rtw_wifi_regd.h>
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 7, 0))
+#define ieee80211_band nl80211_band
+#define IEEE80211_BAND_2GHZ NL80211_BAND_2GHZ
+#define IEEE80211_BAND_5GHZ NL80211_BAND_5GHZ
+#define IEEE80211_NUM_BANDS NUM_NL80211_BANDS
+#endif
+
 static struct country_code_to_enum_rd allCountries[] = {
 	{COUNTRY_CODE_USER, "RD"},
 };
@@ -559,4 +566,4 @@
 
 	_rtw_reg_notifier_apply(wiphy, request, reg);
 }
-#endif //CONFIG_IOCTL_CFG80211
\ No newline at end of file
+#endif //CONFIG_IOCTL_CFG80211
